java/image.bzl

```diff
@@ -114,19 +114,20 @@ def java_files(f):
     Returns:
         Depset of java source files.
     """
-    files = depset()
+    files = []
+
     if java_common.provider in f:
         java_provider = f[java_common.provider]
-        files = depset(transitive = [files, java_provider.transitive_runtime_jars])
-    if hasattr(f, "files"):  # a jar file
```

Comment:
We probably should still keep the check for legacy providers since it hasn't been removed?

Also why changing it back to List? I think it's preferred to use depset. I think it's better for data accumulation like this. https://docs.bazel.build/versions/master/skylark/lib/depset.html

Reply:
Every rule type (even those using the legacy `struct()` instead of providers) expose the `DefaultInfo` provider, so this should be a safe change. See https://docs.bazel.build/versions/master/skylark/lib/DefaultInfo.html

It is preferred to use the depset, I'm only changing how it is constructed. `files` is a list of depsets and then on the last line a new depset is constructed from the list of depsets.

Reply:
Ack. Thanks for the explanation.